core/packages/contracts/contracts/Vault.sol

```diff
@@ -0,0 +1,45 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.9;
+
+import "@openzeppelin/contracts/utils/Address.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+// TODO: transfer ownership from deployer to SovereignTreasury
+// This contract actually holds Ether balances for each sovereignID.
+contract Vault is Ownable {
```

Comment:
Based on OpenZeppelin's [Escrow](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/utils/escrow/Escrow.sol), but stores money per depositer. The vault owner can withdraw the money to any payable address.